CompetitionTemplate/src/competition/subsystems/vision/commands/RotateTowardsBallCommand.java

```diff
@@ -17,42 +20,48 @@
     static Logger log = Logger.getLogger(RotateTowardsBallCommand.class);
     
     protected VisionSubsystem visionSubsystem;
-    private DriveSubsystem driveSubsystem;
+    protected DriveSubsystem driveSubsystem;
+    protected PoseSubsystem poseSubsystem;
+    protected HeadingModule headingModule;
     
-    private PIDManager rotationalPidManager;
+    protected PIDManager rotationalPidManager;
     protected DoubleProperty cameraCenterHeading;
+    protected DoubleProperty currentBallHeadingTarget;
     
     @Inject
-    public RotateTowardsBallCommand(VisionSubsystem visionSubsystem, DriveSubsystem driveSubsystem, XPropertyManager propMan) {
+    public RotateTowardsBallCommand(VisionSubsystem visionSubsystem, DriveSubsystem driveSubsystem, PoseSubsystem poseSubsystem, XPropertyManager propMan, HeadingModule headingModule) {
         this.visionSubsystem = visionSubsystem;
         this.driveSubsystem = driveSubsystem;
+        this.headingModule = headingModule;
         requires(driveSubsystem);
         
-        rotationalPidManager = new PIDManager("Ball rotation", propMan, 0.04, 0, 0);
+        this.headingModule.setPIDPropertyName("Ball rotate");
+        
         cameraCenterHeading = propMan.createEphemeralProperty("Centered cam heading", -15d);
+        currentBallHeadingTarget = propMan.createEphemeralProperty("Current ball target", Double.POSITIVE_INFINITY);
     }
     
     @Override
     public void initialize() {
         log.info("Initializing");
+        
+        BallSpatialInfo targetBall = visionSubsystem.findTargetBall();
+        if(targetBall == null) {
+            log.warn("No target ball found! Nothing to track.");
+            currentBallHeadingTarget.set(Double.POSITIVE_INFINITY);
```

Comment:
If you're going to use a magic number to decide not to do work, then I suggest using a non-infinite one, just in case? Unless you have tested this E2E on the robot system.